consensus/ibft/bridge.go

```diff
@@ -0,0 +1,173 @@
+		if tx.Type != types.TxTypeState {
+			continue
+		}
+
+		if err := b.bridge.ValidateTx(tx); err != nil {
+			b.ibft.logger.Error("block verification failed, block has invalid state transactions", "err", err)
+
+			return errBlockVerificationFailed
+		}
+	}
+
+	return nil
+}
+
+// insertTransactionHookParams are the params passed into the InsertTransactionsHook
+type insertTransactionHookParams struct {
+	header       *types.Header
+	transition   *state.Transition
+	transactions *[]*types.Transaction
+}
+
+// insertStateTransactionsHook applyes state transactions and adds to block
+func (b *BridgeMechanism) insertStateTransactionsHook(rawParams interface{}) error {
+	params, ok := rawParams.(*insertTransactionHookParams)
+	if !ok {
+		return ErrInvalidHookParam
+	}
+
+	msgs, err := b.bridge.GetReadyMessages()
+	if err != nil {
+		b.ibft.logger.Warn("failed to get ready messages from bridge", "err", err)
+
+		return nil
+	}
+
+	signatureThreshold := b.calculateSignatureThreshold(b.ibft.state.validators)
+	for _, msg := range msgs {
+		if uint64(len(msg.Signatures)) < signatureThreshold {
+			b.ibft.logger.Warn(
+				"message doesn't have enough signatures",
+				"wanted", signatureThreshold,
+				"actual", len(msg.Signatures),
+			)
+
+			continue
+		}
+
+		signer := crypto.NewSigner(
+			b.ibft.config.Params.Forks.At(b.ibft.state.view.Sequence),
+			uint64(b.ibft.config.Params.ChainID),
+		)
+
+		tx, err := signer.SignTx(msg.Transaction, b.ibft.validatorKey)
+		if err != nil {
+			return fmt.Errorf("failed to sign state transaction: %w", err)
+		}
+
+		if err := params.transition.Write(tx); err != nil {
+			b.ibft.logger.Warn("failed to apply state transaction", "tx", tx, "err", err)
+
+			continue
+		}
+
+		*params.transactions = append(*params.transactions, tx)
+	}
+
+	return nil
+}
+
+// consumeStateTransactionsHook consumes all state transactions added in the block
+func (b *BridgeMechanism) consumeStateTransactionsHook(numberParam interface{}) error {
+	block, ok := numberParam.(*types.Block)
+	if !ok {
+		return ErrInvalidHookParam
+	}
+
+	for _, tx := range block.Transactions {
+		if tx.Type != types.TxTypeState {
+			continue
+		}
+
+		b.bridge.Consume(tx)
+	}
+
+	return nil
+}
+
+func (b *BridgeMechanism) calculateSignatureThreshold(set ValidatorSet) uint64 {
+	return uint64(set.Len())
+}
+
+// initializeHookMap registers the hooks that the Bridge mechanism
+// should have
+func (b *BridgeMechanism) initializeHookMap() {
+	// Create the hook map
+	b.hookMap = make(map[HookType]func(interface{}) error)
+
+	// Register the AcceptStateLogHook
+	b.hookMap[AcceptStateLogHook] = b.updateValidatorsHook
+
+	// Register the VerifyBlockHook
+	b.hookMap[VerifyBlockHook] = b.verifyStateTransactionsHook
+
+	// Register the InsertTransactionsHook
+	b.hookMap[InsertTransactionsHook] = b.insertStateTransactionsHook
+
+	// Register the InsertBlockHook
+	b.hookMap[InsertBlockHook] = b.consumeStateTransactionsHook
```

Comment:
The hook is called `InsertBlockHook`, but your method is called `consumeStateTransactionsHook`

Reply:
I always like to keep these consistent, but I'll give in now since it's self-explanatory